packages/services/f-braze-adapter/src/services/_tests/BrazeConsumer.test.js

```diff
@@ -0,0 +1,133 @@
+        logger: jest.fn()
+    },
+    brands: [
+        'a',
+        'b',
+        'c',
+        'd'
+    ]
+};
+
+describe('BrazeConsumer', () => {
+    it.each`
+        parameter | value
+        ${'callbacks'} | ${{ 'Consumer callbacks must contain at least ONE callback function': null }}
+        ${'callbacks'} | ${{ 'Consumer callbacks must contain at least ONE callback function': {} }}
+        ${'callbacks'} | ${{ 'Consumer callbacks must be an Object': [] }}
+        ${'interceptInAppMessages'} | ${{ 'Consumer interceptInAppMessages must be an Object': [] }}
+        ${'interceptInAppMessageClickEvents'} | ${{ 'Consumer interceptInAppMessageClickEvents must be an Object': [] }}
+        ${'brands'} | ${{ 'Consumer brands must be an Array': {} }}
+        ${'customFilters'} | ${{ 'Consumer customFilters must be an Array': {} }}
+    `('should throw an error if $parameter is $value', async ({ parameter, value }) => {
+    // Arrange
+    function instantiateBrazeConsumer () {
+        const testOptions = {
+            ...mockConsumerOptions,
+            [parameter]: value[Object.keys(value)[0]]
+        };
+        // eslint-disable-next-line no-unused-vars
+        const consumer = new BrazeConsumer(testOptions);
+    }
+
+    // Assert
+    expect(instantiateBrazeConsumer).toThrow(Object.keys(value)[0]);
+});
+
+    it('should return an array of callbacks from getContentCardCallbacks method', () => {
+        // Arrange
+        const consumer = new BrazeConsumer(mockConsumerOptions);
+
+        // Act
+        const callbacks = consumer.getContentCardCallbacks();
+
+        // Assert
+        expect(callbacks).toEqual(expect.arrayContaining([expect.any(Function)]));
+    });
+
+    it('should call "contentCards" callback when returned callbacks are called', () => {
+        // Arrange
+        const consumer = new BrazeConsumer(mockConsumerOptions);
+        const cards = removeDuplicateContentCards(mockCards.map(transformCardData));
+
+        // Act
+        const callbacks = consumer.getContentCardCallbacks();
+        callbacks.forEach(callback => {
+            callback(cards);
+        });
+
+        // Assert
+        expect(mockConsumerOptions.callbacks.contentCards).toHaveBeenCalled();
+    });
+
+    it('should return an array of callbacks from getInAppMessagesCallbacks method', () => {
+        // Arrange
+        const consumer = new BrazeConsumer(mockConsumerOptions);
+
+        // Act
+        const callbacks = consumer.getInAppMessagesCallbacks();
+
+        // Assert
+        expect(callbacks).toEqual(expect.arrayContaining([expect.any(Function)]));
+    });
+
+    it('should return an array of callbacks from getLoggerCallbacks method', () => {
+        // Arrange
+        const consumer = new BrazeConsumer(mockConsumerOptions);
+
+        // Act
+        const callbacks = consumer.getLoggerCallbacks();
+
+        // Assert
+        expect(callbacks).toEqual(expect.arrayContaining([expect.any(LogService)]));
```

Comment:
Can we check that the correct callbacks are returned?

Reply:
A callback is not returned an object containing callbacks is returned and the type is passed inside of Braze Consumer Registry to fire the correct callback